### PR TITLE
Give windows CI more leeway

### PR DESCRIPTION
**Pull request by kunaltyagi**
Just a test of an idea I had. Might work, might not work :)

### CODE REVIEW

File: .ci/azure-pipelines/build/windows.yaml [hunk 1]
```diff
@@ -27,13 +27,22 @@ steps:
         -DBUILD_surface_on_nurbs=ON
     displayName: 'CMake Configuration'
     workingDirectory: 'c:'
-  - script: |
-      cd %BUILD_DIR% && cmake --build . --config %CONFIGURATION%
-    displayName: 'Build Library'
+  - displayName: 'Build Library'
```
Comment (larshg) [inline .ci/azure-pipelines/build/windows.yaml:30]: displayName needs to come last?

File: .ci/azure-pipelines/build/windows.yaml [hunk 2]
```diff
@@ -27,13 +27,22 @@ steps:
...
+    script: |
+      cd %BUILD_DIR%
+      cmake --build . --config %CONFIGURATION% || ^
+      cmake --build . --config %CONFIGURATION% || ^
+      cmake --build . --config %CONFIGURATION%
     workingDirectory: 'c:'
-  - script: |
-      cd %BUILD_DIR% && cmake --build . --target tests --config %CONFIGURATION%
-    displayName: 'Run Unit Tests'
+  - displayName: 'Run Unit Tests'
```
Comment (larshg) [inline .ci/azure-pipelines/build/windows.yaml:38]: Same here

**Review by mvieth — COMMENTED** [review]
Generally a good idea (and very simple), but
1. failures that are not due to out-of-memory are also rerun
2. would this even work? This is not like rerunning the whole CI, just cmake. I am not sure if running cmake again under the exact same circumstances can lead to success
3. Rerunning the tests is not necessary anyway because they don't fail due to out-of-memory, right?

**Comment by larshg** [comment]
> * failures that are not due to out-of-memory are also rerun
> * would this even work? This is not like rerunning the whole CI, just cmake. I am not sure if running cmake again under the exact same circumstances can lead to success
> * Rerunning the tests is not necessary anyway because they don't fail due to out-of-memory, right?

Just my thoughts about it.

1) Yes, failures will be run again, but they will fail quickly I guess, as it doesn't recompile everything from scratch, but continues on where it failed.

2) Out-of-memory is maybe avoided, if two large required compiler/linking stages are happening simultanously. When rerun, one of them might have completed successfully, so the next time its starts with the one that failed, enough resources are available.

3) There are other tests, that fail sporadically, those that you are adding more printouts for - but maybe we should keep it at one run for the test and hopefully fix those.

**Comment by kunaltyagi** [comment]
1. Yeah, this would increase runtime in those cases because cmake fails on first error. If the 2nd and 3rd cmake don't hit the error, they'd work on compiling other targets not dependent on the failing target (hence increasing run-time).
2. It'd work when we have issues with memory (RAM out-of-memory) but not storage (out-of-space-on-disk)
3. Some tests are flaky. Per lars's recommendation, reduced the re-run count to just the 1

**Comment by mvieth** [comment]
I think since Docker is used for the Windows CI, it has never failed again due to out-of-memory errors or similar. We can observe things for a little bit longer, but if it stays like this, I don't think these changes are necessary.

**Pull request closed without merging**